libraries/stdlib/osl/stdosl.h

```diff
@@ -573,6 +573,169 @@ closure color volume_matte() BUILTIN;
 closure color subsurface(float eta, float g, color mfp, color albedo) BUILTIN;
 #endif
 
+// ******************* MATERIALX PBS LIBRARY CLOSURES - DRAFT BEGIN ******************* //
+
+// -------------------------------------------------------------//
+// BSDF closures                                                //
+// -------------------------------------------------------------//
+​
+// Constructs a diffuse reflection BSDF based on the Oren-Nayar reflectance model.
+// A roughness of 0.0 gives Lambertian reflectance.
+//
+closure color oren_nayar_diffuse_bsdf(normal N, color albedo, float roughness) BUILTIN;
+​
+// Constructs a diffuse reflection BSDF based on the corresponding component of 
+// the Disney Principled shading model.
+//
+closure color burley_diffuse_bsdf(normal N, color albedo, float roughness) BUILTIN;
+​
+// Constructs a reflection and/or transmission BSDF based on a microfacet reflectance
+// model and a Fresnel curve for dielectrics. The two tint parameters control the 
+// contribution of each reflection/transmission lobe. The tints should remain 100% white
+// for a physically correct dielectric, but can be tweaked for artistic control or set
+// to 0.0 for disabling a lobe.
+// The closure may be vertically layered over a base BSDF for the surface beneath the
+// dielectric layer. This is done using the layer() closure. By chaining multiple 
+// dielectric_bsdf closures you can describe a surface with multiple specular lobes.
+// If transmission is enabled (transmission_tint > 0.0) the closure may be layered over
+// a VDF closure describing the surface interior to handle absorption and scattering
+// inside the medium.
+//
+closure color dielectric_bsdf(normal N, vector U, color reflection_tint, color transmission_tint, float roughness_x, float roughness_y, float ior, string distribution) BUILTIN;
+​
+// Constructs a reflection BSDF based on a microfacet reflectance model.
+// Uses a Fresnel curve with complex refraction index for conductors/metals.
+// If an artistic parametrization is preferred the artistic_ior() utility function
+// can be used to convert from artistic to physical parameters.
+//
+closure color conductor_bsdf(normal N, vector U, float roughness_x, float roughness_y, color ior, color extinction, string distribution) BUILTIN;
```

Comment:
Question about the microfacet closures -- are the `roughness_*` arguments meant to be the "alpha" values of the distribution? or is this codifying the `alpha=roughness^2` convention?

It would be good to mention in the comment which values of `distribution` an implementation is expected to support.

Reply:
Not the roughness^2 convention per se, but I'd vote for a linearized one. And the linearization could change with the distribution. Is up to the implementation to do the roughness -> alpha mapping depending on the distribution.

I would decide a reference distribution like GGX with alpha=roughness^2 and try to keep the other distributions roughness response as close to it as possible?

Reply:
About the roughness, in our current implementation the roughness is squared outside, before given to the closure (following vanilla OSL and MDL). In MaterialX we normally connect a node for remapping a `float roughness` and `float anisotropy` into the pair `(roughness_x, roughness_y)` and the `roughness^2` is done as part of this node.

But I agree it would be better to let the implementation handle this, so the given roughness argument is expected to always give a perceptionally linear response, and the caller don't have to care about this.

Reply:
About the microfacet distributions. On our side we have only required GGX so far. In Arnold I think we support GGX and Beckmann, but GGX is the only one used in for example Autodesk Standard Surface. Curious to hear what others have for microfacet distribution needs?

And yes, we should document the set we decide on.

Reply:
We only support GGX at SPI, it was a great simplification of the codebase :) We also use the (roughness, anisotropy) parametrization to the closures.